coverage_map/src/indoor.rs

```diff
@@ -0,0 +1,250 @@
+    }
+}
+
+pub fn insert_indoor_coverage_object(indoor: &mut IndoorCellTree, coverage_object: CoverageObject) {
+    for hex_coverage in coverage_object.coverage.into_iter() {
+        insert_indoor_coverage(
+            indoor,
+            &coverage_object.hotspot_key,
+            &coverage_object.cbsd_id,
+            coverage_object.seniority_timestamp,
+            hex_coverage,
+        );
+    }
+}
+
+pub fn insert_indoor_coverage(
+    indoor: &mut IndoorCellTree,
+    hotspot: &PublicKeyBinary,
+    cbsd_id: &Option<String>,
+    seniority_timestamp: DateTime<Utc>,
+    hex_coverage: UnrankedCoverage,
+) {
+    indoor
+        .entry(hex_coverage.location)
+        .or_default()
+        .entry(hex_coverage.signal_level)
+        .or_default()
+        .push(IndoorCoverageLevel {
+            hotspot_key: hotspot.clone(),
+            cbsd_id: cbsd_id.clone(),
+            seniority_timestamp,
+            signal_level: hex_coverage.signal_level,
+            assignments: hex_coverage.assignments,
+        })
+}
+
+pub fn clone_indoor_coverage_into_submap(
+    submap: &mut IndoorCellTree,
+    from: &IndoorCellTree,
+    coverage_obj: &CoverageObject,
+) {
+    for coverage in &coverage_obj.coverage {
+        if let Entry::Vacant(e) = submap.entry(coverage.location) {
+            if let Some(old_coverage_data) = from.get(&coverage.location) {
+                e.insert(old_coverage_data.clone());
+            }
+        }
+    }
+}
+
+pub fn into_indoor_coverage_map(
+    indoor: IndoorCellTree,
+    boosted_hexes: &impl BoostedHexMap,
+    epoch_start: DateTime<Utc>,
+) -> impl Iterator<Item = RankedCoverage> + '_ {
+    indoor
+        .into_iter()
+        .flat_map(move |(hex, radios)| {
+            let boosted = boosted_hexes.get_current_multiplier(hex, epoch_start);
+            radios
+                .into_values()
+                .flat_map(move |radios| radios.into_sorted_vec().into_iter())
+                .map(move |cov| (hex, boosted, cov))
+        })
+        .enumerate()
+        .map(move |(rank, (hex, boosted, cov))| RankedCoverage {
+            hex,
+            rank: rank + 1,
+            hotspot_key: cov.hotspot_key,
+            cbsd_id: cov.cbsd_id,
+            assignments: cov.assignments,
+            boosted,
+            signal_level: cov.signal_level,
+        })
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::*;
+    use chrono::NaiveDate;
+    use hex_assignments::Assignment;
+    use hextree::Cell;
+
+    #[test]
+    fn ensure_max_signal_level_selected() {
+        let mut indoor_coverage = IndoorCellTree::default();
+        for cov_obj in vec![
+            indoor_cbrs_coverage("1", SignalLevel::None),
+            indoor_cbrs_coverage("2", SignalLevel::Low),
+            indoor_cbrs_coverage("3", SignalLevel::High),
+            indoor_cbrs_coverage("4", SignalLevel::Low),
+            indoor_cbrs_coverage("5", SignalLevel::None),
```

Comment:
Can a `SignalLevel` for an indoor radio be `None`?
When we talked earlier I was led to believe it can only be `High` or `Low`.

Reply:
Yes, it can be High Low or None